wagtail/admin/tests/pages/test_delete_page.py

```diff
@@ -202,3 +221,31 @@ def hook_func(request, page):
 
         # page should be deleted
         self.assertFalse(Page.objects.filter(id=self.child_page.id).exists())
+
+    def test_delete_button_next_url(self):
+        page_perms = PagePerms()
+        page = self.child_page
+        base_url = reverse('wagtailadmin_pages:delete', args=[page.id])
+
+        next_url = "a/random/url/"
+        full_url = base_url + '?' + urlencode({'next': next_url})
+
+        # page_listing_more_button generator yields only `Delete button`
+        delete_button = next(page_listing_more_buttons(
```

Comment:
We shouldn't be testing ``page_listing_more_buttons`` here. We should either test this behaviour at the view level (by asserting on the HTML output or template context), or we should move this into a separate unit test suite for `page_listing_more_buttons`

Reply:
Can I move the tests into wagtail/admin/tests/test_button_hooks.py ?

Reply:
Yep, that sounds good to me, thanks!

Reply:
It's done!

Reply:
awesome, thanks!